themes/default/content/docs/using-pulumi/adopting-pulumi/import/_index.md

```diff
@@ -12,161 +12,184 @@ aliases:
-creating a new one. Using this option lets you specify
-the `import` behavior inside the Pulumi code for your infrastructure deployment, instead of outside of it in a manual workflow.
+Both approaches allow you to adopt and begin managing existing cloud resources with Pulumi, but they work in slightly different ways, and are suited to slightly different use cases. The sections below explain both in more detail.
 
-{{% notes type="warning" %}}
-Your Pulumi stack must be configured correctly---e.g., using the same AWS region as the resource you're importing---otherwise the resource will not be found.
-{{% /notes %}}
+## How resource import works
+
+Import uses the selected stack's configured [provider](/docs/concepts/resources/providers/) to look up the desired resource in the cloud provider, read its current configuration, and add the resource to the stack to bring it under management by Pulumi from that point forward. For this, it requires two pieces of information:
+
+* The _type_ of cloud resource to import --- either as a type _token_ (when using the CLI) or as a resource declaration (when importing in code). The type token of an Amazon S3 Bucket resource, for example, is `aws:s3/bucket:Bucket`.
+
+* The _name_ and _value_ of the property to use for the resource lookup. Lookup properties vary by resource. For an Amazon S3 bucket, the property used for lookup is [`bucket`](/registry/packages/aws/api-docs/s3/bucket/#bucket_nodejs), so the value to use for the lookup would be the bucket's globally unique name. For an Amazon VPC, however, the property used for lookup is [`id`](/registry/packages/aws/api-docs/ec2/vpc/#id_nodejs), so the value to use for it would be its AWS-assigned unique identifier.
+
+### Where to find the type token and lookup property {#where-to-find}
+
+You'll find the type token and lookup property in the Import section of the resource's API documentation in the [Pulumi Registry](/registry/). The type token is quoted in the `pulumi import` example, and the lookup property can be found in the description just above it:
+
+![Where to find the type token and lookup property for a resource](./token-and-lookup.png)
+
+{{< notes >}}
+Make sure the resource provider is configured in a way that allows it to locate the resource you want to import --- e.g., that the resource is in the same region as other resources in your stack --- or the import operation may fail.
+{{< /notes >}}
+
+## Importing resources with the CLI
+
+The `pulumi import` command looks up the resource using the specified type token and resource identifier, adds the resource to the stack's current state, and emits the code required to manage the resource with Pulumi from that point forward. This option requires the least manual effort, so is generally recommended, and is best suited to projects consisting consisting of only one stack.
```

Comment:
I think we need to explain what a type token is. I prefer "type identifier" as it's less jargon-y (if it won't make our docs wildly inconsistent) and may not require much more than an example, e.g. `aws:ec2/vpc:Vpc`, to make it clear what it is (a string that tells Pulumi what resource type it is).

Reply:
Ah yeah, that's better. I'll do that.

Reply:
It looks like we refer to this as "type token" in more places than I realized. Maybe best to keep to that for consistency for now, but it's something worth discussing for sure.